pkg/cache/tas_flavor_snapshot.go

```diff
@@ -344,6 +344,7 @@ func (s *TASFlavorSnapshot) fillInCounts(requests resources.Requests) {
 	lastLevelIdx := len(s.domainsPerLevel) - 1
 	for levelIdx := lastLevelIdx - 1; levelIdx >= 0; levelIdx-- {
 		for _, info := range s.domainsPerLevel[levelIdx] {
+			s.state[info.id] = 0
```

Comment:
Could we avoid reseting the `s.state` in lazy manner? Maybe we could add a dedicated function that would be called after building assignment for a podset

Reply:
Or even better, we could reset/initialize this `s.state` at the beginning of this function

Reply:
Done  -the second option